paddle/phi/kernels/fusion/onednn/fused_transpose_kernel.cc

```diff
@@ -0,0 +1,155 @@
+  }
+
+  in->set_mem_desc(in_md.reshape(squeezed_op_tz));
+  in->Resize(make_ddim(squeezed_op_tz));
+}
+
+void SetOutMemDescWithReshape2FuseSupport(
+    const std::vector<int> fused_reshape2_shape_,
+    phi::DenseTensor* out,
+    const dnnl::memory::desc& out_md) {
+  std::vector<int64_t> fused_reshape2_shape(fused_reshape2_shape_.begin(),
+                                            fused_reshape2_shape_.end());
+
+  const int out_shape_numel = out->numel();
+  const int new_shape_numel = std::accumulate(fused_reshape2_shape.begin(),
+                                              fused_reshape2_shape.end(),
+                                              1,
+                                              std::multiplies<int64_t>());
+
+  for (size_t i = 0; i < fused_reshape2_shape.size(); ++i) {
+    if (fused_reshape2_shape[i] == -1) {
+      fused_reshape2_shape[i] = -out_shape_numel / new_shape_numel;
+      break;
+    }
+  }
+
+  out->set_mem_desc(out_md.reshape(fused_reshape2_shape));
+  out->Resize(phi::make_ddim(fused_reshape2_shape));
+}
+
+template <typename T, typename Context>
+void FusedTransposeKernel(const Context& dev_ctx,
+                          const DenseTensor& x,
+                          const std::vector<int>& axis,
+                          const std::vector<int>& fused_squeeze2_axes,
+                          const std::vector<int>& fused_unsqueeze2_axes,
+                          const std::vector<int>& fused_reshape2_shape,
+                          DenseTensor* out) {
+  PADDLE_ENFORCE_EQ(
+      dev_ctx.GetPlace().GetType(),
+      AllocationType::CPU,
+      errors::PreconditionNotMet("oneDNN Transpose kernel must use CPUPlace"));
+
+  if (!(fused_squeeze2_axes.empty())) {
+    SetInMemDescWithSqueeze2FuseSupport(
+        fused_squeeze2_axes, const_cast<DenseTensor*>(&x), x.mem_desc());
+  }
+
+  if (axis.size() == 1) {
+    Copy<Context>(dev_ctx, x, x.place(), false, out);
+    out->set_mem_desc(x.mem_desc());
+    return;
+  }
+
+  auto x_vec_dims = vectorize(x.dims());
+  auto x_type = funcs::ToOneDNNDataType(x.dtype());
+  funcs::ReorderOneDNNHandler reorder_handler(
+      x_vec_dims, x.dtype(), x_type, dev_ctx.GetEngine());
+  auto reorder_src_memory_p = reorder_handler.AcquireSrcMemory(
+      x.mem_desc(), funcs::to_void_cast(x.data<T>()));
+  auto dst_md =
+      dnnl::memory::desc(x_vec_dims,
+                         x.mem_desc().data_type(),
+                         funcs::GetPlainOneDNNFormat(x_vec_dims.size()));
+
+  // a trick is used here to fake transpose of out_md, so later it will be
+  // "untransposed", leaving output data in plain format tag
+  std::vector<int64_t> fake_strides(axis.size());
+  auto dims = dst_md.dims();
+  int total_stride = 1;
+  for (int i = static_cast<int>(dims.size()) - 1; i >= 0; --i) {
+    fake_strides[axis[i]] = total_stride;
+    total_stride *= dims[axis[i]];
+  }
+  dst_md =
+      dnnl::memory::desc(x_vec_dims, x.mem_desc().data_type(), fake_strides);
```

Comment:
Is it necessary to define `dst_md ` two times? I think that dst_md is not really used after the first definition, as `dst_md.dims() == x_vec_dims.` So it could be changed this way:
```
auto reorder_src_memory_p = reorder_handler.AcquireSrcMemory(
      x.mem_desc(), funcs::to_void_cast(x.data<T>()));
std::vector<int64_t> fake_strides(axis.size());
int total_stride = 1;
for (int i = static_cast<int>(x_vec_dims.size()) - 1; i >= 0; --i) {
    fake_strides[axis[i]] = total_stride;
    total_stride *= x_vec_dims[axis[i]];
  }
auto dst_md = dnnl::memory::desc(x_vec_dims, out_type, fake_strides);
```
I did those changes in this PR in transpose2 kernel: https://github.com/PaddlePaddle/Paddle/pull/49509/files#diff-2f773ea3f0f594fb89c44858d20a3f6f56f45dc55900d669b26c4b3f757f2f69

Reply:
Thank you for spotting this. Fused transpose kernel was implemented before https://github.com/PaddlePaddle/Paddle/pull/49509, so it wasn't up to date.